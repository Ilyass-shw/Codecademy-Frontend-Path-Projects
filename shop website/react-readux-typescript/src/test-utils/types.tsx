import { configureStore, createSlice } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';


export type StoreType = ReturnType<typeof configureStore>;
export type RenderReturn = ReturnType<typeof render>;
export type SliceParameters = Parameters<typeof createSlice>;
/* eslint-disable */
export type UiType = React.ReactElement<
  any,
  string | React.JSXElementConstructor<any>
>;
/* eslint-enable */

export interface TestProviderprops {
  MyStore: StoreType;
  children: /* eslint-disable */
  React.ReactElement<any, string | React.JSXElementConstructor<any>>;
  /* eslint-enable */
}
