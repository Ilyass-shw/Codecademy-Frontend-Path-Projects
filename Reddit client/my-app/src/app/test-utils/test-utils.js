import React from "react";
import { render as rtlRender } from "@testing-library/react";
// import { createStore } from "redux";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

let Store;
Store = mockStore({});

function render(ui, { initialState, store = Store, ...renderOptions } = {}) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
