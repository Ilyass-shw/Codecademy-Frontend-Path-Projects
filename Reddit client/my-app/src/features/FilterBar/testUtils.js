import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../app/store'

const TestProvider = ({
  store,
  children
}) => <Provider store={store}>{children}</Provider>

export function render(ui, { store, ...otherOpts }) {
  return rtlRender(<TestProvider store={store}>{ui}</TestProvider>, otherOpts)
}

export function makeTestStore(opts = {}) {
  const origDispatch = store.dispatch
  store.dispatch = jest.fn(origDispatch)
  return store
}

// re-export everything
export * from "@testing-library/react";

