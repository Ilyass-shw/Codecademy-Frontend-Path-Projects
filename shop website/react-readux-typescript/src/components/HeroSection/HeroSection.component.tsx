import styled from 'styled-components';

export const BackgroundImage = styled.section`
  background: url('https://jonnajintonsweden.com/wp-content/uploads/2020/09/test-header-4-4500x1550.jpg')
    45% / cover no-repeat;
  height: 22.2rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 6rem;

  @media screen and (min-width: 520px) {
    background-position: 25% 25%;
  }
`;

// ============= ============= ============= ============= =============

export const HeroContent = styled.div`
  width: 40vw;
  height: 60%;
  color: #f1f1f1;
`;

// ============= ============= ============= ============= =============

export const HeroText = styled.div`
  @import url('https://fonts.googleapis.com/css2?family=Quattrocento+Sans:ital@1&display=swap');

  p {
    padding: 10px;
    font-family: 'Quattrocento Sans', sans-serif;
    font-style: italic;
    font-size: 14px;
  }
`;

export const FirstPart = styled.p``;

// ============= ============= ============= ============= =============

export const LastPart = styled.p``;

// ============= ============= ============= ============= =============
