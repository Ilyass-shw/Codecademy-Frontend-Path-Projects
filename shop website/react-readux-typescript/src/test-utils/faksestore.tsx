// import { configureStore } from '@reduxjs/toolkit';
// import { createSlice } from '@reduxjs/toolkit';

// import { StoreType , SliceParameters} from './types';

// // const initialState :object= {};
// const fakeSlice = createSlice({
//   name: 'posts',
//   initialState,
//   reducers: {},
// });

// export default configureStore({
//   reducer: {
//     posts: fakeSlice.reducer,
//   },
// });

// export const makeCustomTestStore = (
//   name: string,
//   initialState: string[],
//   reducers = {},
// ): StoreType => {
//   const fakeSlice = createSlice({
//     name,
//     initialState,
//     reducers,
//   });

//   return configureStore({
//     reducer: {
//       posts: fakeSlice.reducer,
//     },
//   });
// };
