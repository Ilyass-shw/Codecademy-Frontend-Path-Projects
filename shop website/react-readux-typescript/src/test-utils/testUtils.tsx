import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import store, { makeCustomTestStore } from './faksestore';
import { RenderReturn, StoreType } from './types';
import { BrowserRouter } from 'react-router-dom';

// const TestWrapper: React.FC<WrapperParams> = ({
//   MyStore,
//   WithRouter,
//   children,
// }) => {
//   if (WithRouter) {
//     return (
//       <Provider store={MyStore}>
//         <BrowserRouter>{children}</BrowserRouter>
//       </Provider>
//     );
//   } else {
//     return <Provider store={MyStore}>{children}</Provider>;
//   }
// };

// ============= ============= ============= ============= =============

export const render = (
  ui: React.ReactElement,
  MyStore: StoreType,
  WithRouter?: 'withRouter',
): RenderReturn => {
  if (WithRouter) {
    return rtlRender(
      <Provider store={MyStore}>
        <BrowserRouter>{ui}</BrowserRouter>
      </Provider>,
    );
  } else {
    return rtlRender(<Provider store={MyStore}>{ui}</Provider>);
  }
};

// ============= ============= ============= ============= =============

export const renderWithRouterOnly = (ui: React.ReactElement): RenderReturn => {
  return rtlRender(<BrowserRouter>{ui}</BrowserRouter>);
};

// ============= ============= ============= ============= =============

export const makeTestStore = (
  Name?: string,
  initialState?: Record<string, unknown>,
  reducers = {},
): StoreType => {
  if (Name && initialState) {
    const CustomTestStore = makeCustomTestStore(Name, initialState, reducers);
    const origDispatch = CustomTestStore.dispatch;
    CustomTestStore.dispatch = jest.fn(origDispatch);
    return CustomTestStore;
  } else {
    const origDispatch = store.dispatch as jest.Mock;
    store.dispatch = jest.fn(origDispatch);
    return store;
  }
};

// re-export everything
export * from '@testing-library/react';
