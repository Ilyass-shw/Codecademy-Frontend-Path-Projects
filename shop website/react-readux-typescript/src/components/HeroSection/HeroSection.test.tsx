import React from 'react';
import { render } from '../../test-utils/testUtils';
import HeroSection from './HeroSection';

describe('NavBar', () => {
  it('should render', () => {
    render(<HeroSection />);
  });
  it('should match snapshot', () => {
    const { asFragment } = render(<HeroSection />);
    expect(asFragment()).toMatchSnapshot();
  });
});
