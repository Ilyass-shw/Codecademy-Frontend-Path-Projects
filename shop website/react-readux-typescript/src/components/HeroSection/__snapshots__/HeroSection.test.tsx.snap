// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-bdvvaa lchavo"
  >
    <div
      class="sc-gsDJrp gfkuTp"
    >
      <div
        class="sc-dkPtyc bsZSCH"
      >
        <p
          class="sc-hKwCoD"
        >
          “Art is like a dance of the Heart,
        </p>
        <p
          class="sc-eCImvq"
        >
           when the Soul is free…”
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
